src/main/java/com/dain_review/domain/post/service/FollowService.java

```diff
@@ -0,0 +1,234 @@
+                        .user(user)
+                        .categoryType(followRequest.categoryType())
+                        .title(followRequest.title())
+                        .content(followRequest.content())
+                        .followType(followRequest.followType())
+                        .build();
+
+        PostMeta postMeta =
+                PostMeta.builder()
+                        .post(post)
+                        .viewCount(0L) // 조회수와 댓글수는 0으로 초기화
+                        .commentCount(0L)
+                        .build();
+
+        post.setPostMeta(postMeta);
+        postRepository.save(post);
+
+        // 이미지 저장 및 url 반환
+        saveImageFiles(imageFiles, post);
+        List<String> imageUrls = findImageUrls(post.getId());
+
+        return FollowResponse.responseWithoutContentPreview(post, imageUrls);
+    }
+
+    public FollowResponse updatePost(
+            Long userId, Long postId, FollowRequest followRequest, List<MultipartFile> imageFiles
+    ) {
+        User user = getUser(userId);
+
+        Post existingPost =
+                postRepository
+                        .findById(postId)
+                        .orElseThrow(() -> new PostException(PostErrorCode.POST_NOT_FOUND));
+
+        if (!existingPost.getUser().getId().equals(user.getId())) {
+            throw new PostException(PostErrorCode.UNAUTHORIZED_ACCESS);
+        }
+
+        PostMeta existingPostMeta = existingPost.getPostMeta();
+
+        Post updatedPost =
+                Post.builder()
+                        .id(existingPost.getId())
+                        .user(existingPost.getUser())
+                        .categoryType(existingPost.getCategoryType())
+                        .title(followRequest.title()) // 새 제목 설정
+                        .content(followRequest.content()) // 새 내용 설정
+                        .followType(followRequest.followType())
+                        .createdAt(existingPost.getCreatedAt())
+                        .updatedAt(LocalDateTime.now()) // 업데이트 시각 설정
+                        .postMeta(existingPostMeta)
+                        .build();
+
+        postRepository.save(updatedPost);
+
+        // 새로 추가된 이미지 저장
+        saveImageFiles(imageFiles, updatedPost);
+        // 기존 이미지 중 삭제된 이미지 파일명 리스트를 전달받아 해당하는 파일 삭제처리
+        deleteImageFiles(followRequest.deleted());
+        List<String> imageUrls = findImageUrls(updatedPost.getId());
+
+        return FollowResponse.responseWithoutContentPreview(updatedPost, imageUrls);
+    }
+
+    public void deletePost(Long userId, Long postId) {
+        User user = getUser(userId);
+
+        Post post =
+                postRepository
+                        .findById(postId)
+                        .orElseThrow(() -> new PostException(PostErrorCode.POST_NOT_FOUND));
+
+        if (!post.getUser().getId().equals(user.getId())) {
+            throw new PostException(PostErrorCode.UNAUTHORIZED_ACCESS);
+        }
+
+        post.setDeleted(true);
```

Comment:
public void deleteBy(User user){
    if(isNotWriter(user)) {
         throw new PostException(PostErrorCode.UNAUTHORIZED_ACCESS);
   }
    this.deleted = true;
    this.deletedAt = now();
};

public boolean isWriter(User user) {. 
    this.user.isSame(user);
}

public boolean isNotWriter(User user) {. 
   return !this.isWriter(user);
}